File: LeetCode/1470. Shuffle the Array/solution.go
/**
 * 53 / 53 test cases passed.
 * Runtime: 8 ms
 * Memory Usage: 3.6 MB 
 */
 func shuffle(nums []int, n int) []int {
    ret := make([]int, 0, n + n)
    for i := 0; i < n; i ++ {
        ret = append(ret, nums[i], nums[i + n])
    }
    return ret
}

File: LeetCode/1470. Shuffle the Array/solution.py
"""
53 / 53 test cases passed.
Runtime: 24 ms
Memory Usage: 15.2 MB
"""
class Solution:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        return [nums[(i >> 1) + n] if i & 1 else nums[(i >> 1)] for i in range(2 * n)]


"""
53 / 53 test cases passed.
Runtime: 50 ms
Memory Usage: 15.1 MB
"""
class Solution2:
    def shuffle(self, nums: List[int], n: int) -> List[int]:
        return sum(zip(nums[:n], nums[n:]), ())
    